File: java/pocs/src/main/java/io/antmendoza/samples/_5611/Workflow_5611Impl.java
package io.antmendoza.samples._5611;

import io.temporal.activity.ActivityOptions;
import io.temporal.workflow.Workflow;

import java.time.Duration;

public class Workflow_5611Impl implements Workflow_5611 {

    private final Activity_5611 activity = Workflow.newActivityStub(Activity_5611.class,
            ActivityOptions.newBuilder().
                    setStartToCloseTimeout(Duration.ofSeconds(20))
                    .build()
    );


    @Override
    public String run() {


        boolean activityCompleted = Workflow.await(Duration.ofSeconds(5), () -> {
            System.out.println("In Workflow.await java");
            activity.doSomething();
            return true;
        });

        System.out.println("Completed " + !activityCompleted );

        if (!activityCompleted) {
            System.out.println(">>>>>>>>>  timer fired");
        }

        return "done";
    }
}

File: kotlin/_5611/src/main/kotlin/Workflow_5611Impl.kt
import io.temporal.activity.ActivityOptions
import io.temporal.workflow.Workflow
import java.time.Duration

class Workflow_5611Impl : Workflow_5611 {

    private val signalValue = false

    private val activity = Workflow.newActivityStub(
        Activity_5611::class.java,
        ActivityOptions.newBuilder().
        setStartToCloseTimeout(Duration.ofSeconds(20))
            .build()
    )

    override fun run(): String {


//        val awaitSignal = Workflow.await(Duration.ofSeconds(5)) { signalValue }
        val awaitSignal = Workflow.await(Duration.ofSeconds(5)) {
            println("In Workflow.await")
            activity.doSomething()
            signalValue
        }



        if (!awaitSignal) {
            println("timer fired")
        }

        return "done"
    }

}
